barretenberg/cpp/src/barretenberg/client_ivc/mock_circuit_producer.hpp

```diff
@@ -183,18 +183,17 @@ class ClientIVCMockCircuitProducer {
      * only necessary if the structured trace is not in use).
      *
      */
-    static ClientCircuit create_mock_circuit(ClientIVC& ivc, size_t log2_num_gates = 16)
+    static ClientCircuit create_mock_circuit(ClientIVC& ivc, size_t log2_num_gates = 16, bool is_kernel = false)
     {
-        ClientCircuit circuit{ ivc.goblin.op_queue };
+        ClientCircuit circuit{ ivc.goblin.op_queue, is_kernel };
         MockCircuits::construct_arithmetic_circuit(circuit, log2_num_gates, /* include_public_inputs= */ false);
         return circuit;
     }
 
   public:
     ClientCircuit create_next_circuit(ClientIVC& ivc, size_t log2_num_gates = 16, const size_t num_public_inputs = 0)
     {
```

Comment:
this was actually a kinda bug cus create_mock_circuit was recreating the circuit